### PR TITLE
Feature/issue 113 notifications

### PR DESCRIPTION
**Pull request by bensinjin**
A couple things to note:

- This PR includes what I'm referring to as "Expiring notifications" that are removed after N seconds
- I've opted to keep timer state in the component as to me it felt that's where it belonged, not in Redux
- Currently you will continue to see the same "Task added to my plan" notification when clicking the add button to tasks already in your plan. This works as expected but in the future we will not be displaying the "+" button on tasks already in your plan. This was one of the requests produced from recent testing sessions.

### CODE REVIEW

File: src/components/notifications/expiring_notification.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        }, this.props.timeInSeconds * 1000);
+    }
+
+    componentWillUnmount(): void {
+        clearTimeout(this.timer);
+    }
+
+    render(): JSX.Element {
+        return(
+            <Text style={[{color: colors.white}]}><Trans>{this.props.notification.text}</Trans></Text>
```
Comment (tnightingale) [inline src/components/notifications/expiring_notification.tsx]: Don't put variables in `<Trans>`, it doesn't do anything. Only string literals and/or other React components containing string literals.

File: src/stores/notifications.ts [hunk 2]
```diff
@@ -36,13 +47,26 @@ export const reducer = (store: Store = buildDefaultStore(), action?: Notificatio
...
+                    [action.payload.notification.id]: action.payload.notification,
+                },
             };
-        case constants.REMOVE_LAST_NOTIFICATION:
+        case constants.REMOVE_NOTIFICATION:
+            const isNotActionNotification = (notification: Notification): boolean =>
+                notification.id !== action.payload.notificationId;
             return {
-                notifications: R.drop(1, store.notifications),
+                notifications: R.pickBy(isNotActionNotification, store.notifications),
```
Comment (rasmus-storjohann-PG) [inline src/stores/notifications.ts]: There is also `R.propEq()`

**Comment by rasmus-storjohann-PG** [comment]
I reviewed commit by commit, but went through and updated comments that were addressed later in the PR.

File: src/stores/notifications.ts (hunk 2 above)
Comment (rasmus-storjohann-PG) [reply src/stores/notifications.ts]: See https://stackoverflow.com/questions/35921591/apply-logical-not-to-propeq-and-filter-in-ramda/35922399#35922399

File: src/components/notifications/expiring_notification.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    readonly notification: Notification;
+    readonly timeInSeconds: number;
+}
+export interface ExpiringNotificationActions {
+    readonly timeElapsedCallback: () => void;
+}
+type AllExpiringNotificationProps = ExpiringNotificationProps & ExpiringNotificationActions;
+
+export class ExpiringNotificationComponent extends React.Component<AllExpiringNotificationProps> {
+    timer: number;
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/components/notifications/expiring_notification.tsx]: Yeah, it's a numeric id.

File: src/components/notifications/expiring_notification.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+export interface ExpiringNotificationProps {
+    readonly notification: Notification;
+    readonly timeInSeconds: number;
+}
+export interface ExpiringNotificationActions {
+    readonly timeElapsedCallback: () => void;
+}
+type AllExpiringNotificationProps = ExpiringNotificationProps & ExpiringNotificationActions;
+
+export class ExpiringNotificationComponent extends React.Component<AllExpiringNotificationProps> {
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/components/notifications/expiring_notification.tsx]: My initial thought was to use a "withExpiration" HOC but struggled to get it playing nice with TypeScript.

File: src/components/notifications/notifications.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                );
+            }, R.keys(props.notifications))}
+        </View>
+    );
+};
+
+const renderNotification = (notification: Notification, props: AllNotificationsProps): JSX.Element => {
+    switch (notification.type) {
+        default:
+        case NotificationType.Expiring:
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/components/notifications/notifications.tsx]: Good point.

File: src/stores/__tests__/notifications.test.ts [hunk 6]
```diff
@@ -1,38 +1,55 @@
...
             const finalStore = model.reducer(store, addToSavedList(task.id));
-            expect(finalStore.notifications.length).toBe(3);
+            const lastKey = R.keys(finalStore.notifications)[2];
+            expect(finalStore.notifications[lastKey].type).toBe(model.NotificationType.Expiring);
         });
 
-        it ('allows for the removal of the last notificiation', () => {
-            const finalStore = model.reducer(store, model.removeLastNotification());
-            expect (finalStore.notifications.length).toBe(1);
+        it ('allows for the removal of a notificiation', () => {
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply src/stores/__tests__/notifications.test.ts]: Fixed

File: src/stores/notifications.ts (hunk 2 above)
Comment (bensinjin) [reply src/stores/notifications.ts]: Sweet, fixed.

**Pull request merged**